server/app/views/components/Icons.java

```diff
@@ -312,34 +366,30 @@ public static SvgTag questionTypeSvg(QuestionType type, int width, int height) {
       default:
         icon = Icons.UNKNOWN;
     }
-    return svg(icon, width, height);
+    return svg(icon, size);
   }
 
-  public static SvgTag svg(Icons icon, int pixelSize) {
-    return svg(icon.path, pixelSize);
-  }
-
-  public static SvgTag svg(String pathString, int pixelSize) {
-    return svg(pathString, pixelSize, pixelSize);
-  }
-
-  public static SvgTag svg(Icons icon, int width, int height) {
-    return svg(icon.path, width, height);
-  }
-
-  public static SvgTag svg(String pathString, int width, int height) {
+  /** Don't use it. Use {@link #questionTypeSvg(QuestionType type) instead} */
+  public static SvgTag svg(Icons icon, int size) {
+    if (icon.size != size) {
+      throw new IllegalArgumentException(
```

Comment:
Our ability to detect this is only as good as our test coverage. Given that you'll be converting so that this type of error isn't possible in a future PR, can we downgrade this to a log statement to prevent a gap in our test coverage from potentially causing a user-facing error that is recoverable?

Reply:
This check was useful for me to catch errors in existing code by running test. I agree that having exceptions in prod will cause more problems than preventing actual major issue. Downgraded to logging.